scheduler/pkg/kafka/config/tls.go

```diff
@@ -18,39 +18,75 @@ package config
 
 import (
 	"github.com/confluentinc/confluent-kafka-go/kafka"
+	"github.com/seldonio/seldon-core/components/tls/v2/pkg/password"
 	"github.com/seldonio/seldon-core/components/tls/v2/pkg/tls"
+	"github.com/seldonio/seldon-core/components/tls/v2/pkg/util"
 )
 
 const (
-	EnvKafkaClientPrefix = "KAFKA_CLIENT"
-	EnvKafkaBrokerPrefix = "KAFKA_BROKER"
+	EnvKafkaClientPrefix      = "KAFKA_CLIENT"
+	EnvKafkaBrokerPrefix      = "KAFKA_BROKER"
+	EnvSASLUsernameSuffix     = "_SASL_USERNAME"
+	EnvPasswordLocationSuffix = "_SASL_PASSWORD_LOCATION"
+	DefaultSASLUsername       = "seldon"
 )
 
 func AddKafkaSSLOptions(config kafka.ConfigMap) error {
-	var cs *tls.CertificateStore
-	var err error
 	protocol := tls.GetSecurityProtocolFromEnv(tls.EnvSecurityPrefixKafka)
-	if protocol == tls.SecurityProtocolSSL {
-		cs, err = tls.NewCertificateStore(tls.Prefix(EnvKafkaClientPrefix),
-			tls.ValidationPrefix(EnvKafkaBrokerPrefix))
-		if err != nil {
-			return err
-		}
+	switch protocol {
+	case tls.SecurityProtocolSSL:
+		return setupTLSAuthentication(config)
+	case tls.SecurityProtocolSASLSSL:
+		return setupSASLSSLAuthentication(config)
+	case tls.SecurityProtocolPlaintxt:
+		return nil
 	}
-	if cs != nil {
-		cert := cs.GetCertificate()
-		caCert := cs.GetValidationCertificate()
-		config["security.protocol"] = "ssl"
-		// issue is that ca.pem does not work with multiple certificiates defined
-		// see https://github.com/confluentinc/confluent-kafka-go/issues/827
-		if caCert != nil {
-			config["ssl.ca.location"] = caCert.CaPath
-		} else {
-			config["ssl.ca.location"] = cert.CaPath
-		}
-		config["ssl.key.location"] = cert.KeyPath
-		config["ssl.certificate.location"] = cert.CrtPath
-		config["ssl.endpoint.identification.algorithm"] = "none"
+	return nil
+}
+
+func setupSASLSSLAuthentication(config kafka.ConfigMap) error {
+	cs, err := tls.NewCertificateStore(tls.ValidationOnly(true), tls.ValidationPrefix(EnvKafkaBrokerPrefix))
+	if err != nil {
+		return err
+	}
+	caCert := cs.GetValidationCertificate()
+	config["security.protocol"] = "SASL_SSL"
+	config["sasl.mechanism"] = "SCRAM-SHA-512"
+	ps, err := password.NewPasswordStore(password.Prefix(EnvKafkaClientPrefix),
+		password.LocationSuffix(EnvPasswordLocationSuffix))
+	if err != nil {
+		return err
+	}
+	username, found := util.GetEnv(EnvKafkaClientPrefix, EnvSASLUsernameSuffix)
+	if !found {
+		username = DefaultSASLUsername
+	}
+	config["sasl.username"] = username
+	config["sasl.password"] = ps.GetPassword()
+	// issue is that ca.pem does not work with multiple certificiates defined
+	// see https://github.com/confluentinc/confluent-kafka-go/issues/827 (Fixed but not yet released)
```

Comment:
apparently this is now fixed as per [comment](https://github.com/confluentinc/confluent-kafka-go/issues/827#issuecomment-1303413207)?

Reply:
Maybe leave to a future PR and upgrade of librdkafka we use.